### ISSUE TITLE
Broken dependency

### ISSUE THREAD

**Issue by Youngjae-KAIST** [DigiRL-agent/digiq#16]
This could be related to the issue #14 

```
git clone git@github.com:haotian-liu/LLaVA.git
cd LLaVA
conda create -n llava python=3.10 -y
conda activate llava
pip install --upgrade pip  # enable PEP 660 support
pip install -e .
pip install -e ".[train]"
pip install flash-attn --no-build-isolation
pip install gradio==4.43.0 # important
```
This code tries installing latest flash-attn which is incompatible with llava.
I tried updating to other version

```
pip install flash-attn==2.7.2.post1
```

Then, other dependency breaks.

vlm_finetune runs on following versions although I am not sure about loss.

transformers 4.51.0
accelerate 1.6.0
deepspeed 0.16.5
torch 2.6.0
flash-attn 2.5.8 or 2.5.9.post1

gradio_based vlm_infer works on 2.5.9.post1 version of flash_attn and 5.23.3 version of gradio (without upgrading other libraries) .
otherwise gradio would throw an error.

vllm_based vlm_infer did not work for me.

```
wget https://github.com/vllm-project/vllm/archive/d98929a23940aa3e8f1d3544fe4ef6e1c7838e3c.zip # from https://github.com/vllm-project/vllm/pull/7892/commits
unzip d98929a23940aa3e8f1d3544fe4ef6e1c7838e3c.zip
mv vllm-d98929a23940aa3e8f1d3544fe4ef6e1c7838e3c vllm

conda create --name vllm python=3.10.0 -y
conda activate vllm
pip install -e vllm # this will take 15-30min depending on your CPU speed
```

This would throw
```
ptxas tmp/CMakeCUDACompilerId.ptx, line 9; fatal : Unsupported .version
        8.7; current version is '8.4'
```
on cuda12.4, Ubuntu 22.04.5 LTS

Could you specify the version of flash_attn and give a cuda version for vllm.
I have been working on this for few days so
I would appreciate if you give a guide about this problem.

**Comment by BiEchi** [comment]
I would suggest using `gradio==4.43.0`, we met the gradio version problems for several times during development. 

Usually, as long as you can run the training (`finetune.sh`), the loss you obtained is accurate, so if the loss is really low there's probably something wrong with your data, instead of training pipeline.

For flash-attn compatibility with LLaVA, you should refer to the flash-attn issues ([this issue](https://github.com/Dao-AILab/flash-attention/issues/1421) might help) and the liuhaotian/llava issues ([this issue](https://github.com/haotian-liu/LLaVA/issues/1248) should help). The packages version are shown in [the requirement file](https://github.com/DigiRL-agent/digiq/blob/main/requirements.txt).

For LLaVA+gradio, we find no problem using gradio==4.43.0 + the latest version of LLaVA.

**Comment by Youngjae-KAIST** [comment]
>For flash-attn compatibility with LLaVA, you should refer to the flash-attn issues (https://github.com/Dao-AILab/flash-attention/issues/1421 might help) and the liuhaotian/llava issues (https://github.com/haotian-liu/LLaVA/issues/1248 should help)

Unfortunately, it did not solve dependency issue of mine.
I think I would look for the issues in their repo.

Thank you for your response!
